src/encoding.md

```diff
@@ -225,6 +226,49 @@ to be percent-encoded. The choice of this set depends on context. For example,
 The return value of encoding is an iterator of `&str` slices which can be
 collected into a `String`.
 
+[ex-base64]: #ex-base64
+<a name="ex-base64"></a>
+## Encode and decode base64
+
+[![base64-badge]][base64] [![cat-encoding-badge]][cat-encoding]
+
+The [`base64`] crate provides `encode` and `decode` functions.
+The example below encode a byte slice to a `base64` encoded `String`
+then decode it. The results are printed to the standard output.
+
+[`base64`]: https://docs.serde.rs/base64/
+
+```rust
+#[macro_use]
+extern crate error_chain;
+
+extern crate base64;
+
+use std::str;
+use base64::{encode, decode};
+
+error_chain! {
+    foreign_links {
+        Base64(base64::DecodeError);
+        Utf8Error(str::Utf8Error);
+    }
+}
+
+fn run() -> Result<()> {
+    let hello = b"hello rustaceans";
+    let encoded = encode(hello);
+    let decoded = &decode(&encoded)?[..];
```

Comment:
I would suggest 
let decoded = decode(&encoded)?;
println!("back to origin: {}", str::from_utf8(&decoded)?);